contrib/opentimelineio_contrib/application_plugins/rv/example_otio_writer/otio_writer.py

```diff
@@ -0,0 +1,372 @@
+                       cut_in_frame=None):
+    """
+    Create an OTIO Transition for a CrossDissolve node.
+    :param node_name: `str`
+    :return: `otio.schema.Transition`
+    """
+    transition_type = TRANSITION_TYPE_MAP.get(
+        rv.commands.nodeType(node_name),
+        otio.schema.transition.TransitionTypes.Custom
+    )
+    if not transition_type:
+        return None
+
+    input_node_names, _ = rv.commands.nodeConnections(node_name)
+
+    # Assume FPS matches first input (RV Transition don't have FPS settings)
+    fps = get_source_fps(input_node_names[0])
+
+    durations_frames = rv.commands.getFloatProperty(
+        node_name + ".parameters.numFrames")
+    if not durations_frames:
+        duration_frames = 20.0
+    else:
+        duration_frames = durations_frames[0]
+
+    in_offset_frames = int(math.floor(duration_frames/2))
+    in_offset = otio.opentime.RationalTime(in_offset_frames, rate=fps)
+    out_offset = otio.opentime.RationalTime(duration_frames-in_offset_frames,
+                                            rate=fps)
+
+    transition = otio.schema.Transition(
+        name=get_node_ui_name(node_name),
+        transition_type=transition_type,
+        in_offset=in_offset,
+        out_offset=out_offset,
+        metadata=get_node_otio_metadata(node_name)
+    )
+    return transition
+
+
+def get_node_ui_name(node_name):
+    """
+    Retrieve the value of the UI Name property for a node, if it exists.
+    :param node_name: `str`
+    :return: `str`
+    """
+    prop_name = node_name + ".ui.name"
+    if rv.commands.propertyExists(prop_name):
+        return rv.commands.getStringProperty(prop_name)[0]
+
+    return None
+
+
+def get_node_otio_metadata(node_name):
+    """
+    Retrieve the value of the OTIO Metadata property for a node, if it exists.
+    This is set by otio_reader.py application plugin.
+    :param node_name: `str`
+    :return: `str`
+    """
+    prop_name = node_name + ".attributes.otio_metadata"
+    if rv.commands.propertyExists(prop_name):
+        metadata_str = rv.commands.getStringProperty(prop_name)[0]
+        try:
+            metadata = json.loads(metadata_str)
+        except TypeError:
+            logging.error("{} not JSON compatible type ('{}')".format(
+                prop_name, type(metadata_str)))
+        except ValueError:
+            logging.error("{} not JSON ({}: '{}')".format(
+                prop_name, type(metadata_str), metadata_str))
+        else:
+            return metadata
+
+    return None
+
+
+def frames_to_rational_times(start_frame, end_frame, fps):
+    """
+    Convert the start and end frames and fps to OTIO Rational Times.
+    :param start_frame: `int`
+    :param end_frame: `int`
+    :param fps: `float`
+    :return: (`opentimelineio.opentime.RationalTime`, `opentimelineio.opentime.RationalTime`)
+    """
+    if start_frame is None or end_frame is None or not fps:
+        return None, None
+
+    duration = end_frame - start_frame + 1
```

Comment:
There's a `range_from_start_end_time` convenience function you can use to avoid this computation.

Reply:
Thanks.